dev/src/pool.ts

```diff
@@ -195,6 +205,14 @@ export class ClientPool<T> {
 
   async terminate(): Promise<void> {
     this.terminated = true;
+
+    // If there are any operations still running after terminate() is called,
```

Comment:
I think this only thing you have to do here is set `maxIdleClients` to 0 and call `shouldGarbageCollectClient()`.

It will remove the remaining client if it  is currently idle. If not, the next run of `shouldGarbageCollectClient()` will trigger the removal (which should happen after `release()`)